### ISSUE TITLE
Refactor transformer to use the instance of `AstBuilder` which is in `TraverseCtx` everywhere

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#6172]
`AstBuilder` is the object which you use to generate AST nodes. It is heavily used in the transformer.

We currently have 2 copies of `AstBuilder` in transformer:

* 1st copy is stored in `TransformCtx`
* 2nd copy is stored in `TraverseCtx`.

This is problematic for 3 reasons:

1. It blocks us from doing #6171, which will be a performance boost.
2. For the introduction of AST node IDs (#5689), `AstBuilder` will become stateful, containing a counter for creating unique IDs. So only a single copy of it can exist.
3. It's rather messy! Why do we have 2 copies anyway?

We need to go through the transformer and swap all uses of `AstBuilder` from `TransformCtx` over to use the copy in `TraverseCtx`.

For example:

https://github.com/oxc-project/oxc/blob/1cee2070506ec58a8dea86ab9ff3ca3258a98ac3/crates/oxc_transformer/src/react/jsx_source.rs#L92-L96

Should become:

```rs
let key = ctx.ast.property_key_identifier_name(SPAN, SOURCE); 
let value = self.get_source_object(line, column, ctx); 
ctx.ast.object_property_kind_object_property(SPAN, kind, key, value, None, false, false, false)
```

This will require passing `TraverseCtx` into some functions which it's currently not available in.

I have marked this as "good first issue" as it's not too technically tricky (but then again, there are always some unforeseen complications!), but is a good opportunity for anyone interested in the transformer to get to know the codebase.

Would anyone be able to help please?

**Comment by shulaoda** [comment]
🙌 I'm happy to

**Comment by shulaoda** [comment]
I found that there is also an `AstBuilder` instance in `module_imports` of `TransformCtx`. Does this need improvement?

**Comment by Dunqing** [comment]
> I found that there is also an `AstBuilder` instance in `module_imports` of `TransformCtx`. Does this need improvement?

Leave `module_imports` for me, I am going to refactor it.

**Comment by overlookmotel** [comment]
Closed by #6173 and #6209. Thanks shulaoda!
